tests/Set/SymmetricDifferenceTest.php

```diff
@@ -0,0 +1,638 @@
+<?php
+
+declare(strict_types=1);
+
+namespace IterTools\Tests\Set;
+
+use IterTools\Set;
+use IterTools\Tests\Fixture\ArrayIteratorFixture;
+use IterTools\Tests\Fixture\GeneratorFixture;
+use IterTools\Tests\Fixture\IteratorAggregateFixture;
+
+class SymmetricDifferenceTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @dataProvider dataProviderForArray
+     * @param        array<array> $iterables
+     * @param        array $expected
+     */
+    public function testArray(array $iterables, array $expected): void
+    {
+        // Given
+        $result = [];
+
+        // When
+        foreach (Set::symmetricDifference(...$iterables) as $datum) {
+            $result[] = $datum;
+        }
+
+        // Then
+        $this->assertEquals($expected, $result);
+    }
+
+    public function dataProviderForArray(): array
+    {
+        return [
+            [
+                [],
+                [],
+            ],
+            [
+                [
+                    [],
+                ],
+                [],
+            ],
+            [
+                [
+                    [],
+                    [],
+                ],
+                [],
+            ],
+            [
+                [
+                    [],
+                    [],
+                    [],
+                ],
+                [],
+            ],
+            [
+                [
+                    [1, 2, 3, 4, 5],
+                    [1, 2, 3, 4, 5],
+                ],
+                [],
+            ],
+            [
+                [
+                    [1, 2, 3, 4, 5],
+                    [1, 2, 3, '4', '5'],
+                ],
+                [4, '4', 5, '5'],
+            ],
+            [
+                [
+                    ['1', '2', '3', 4, 5],
+                    [1, 2, 3, '4', '5'],
+                ],
+                ['1', 1, '2', 2, '3', 3, 4, '4', 5, '5'],
+            ],
+            [
+                [
+                    [1, 1, 2, 2, 1, 1],
+                    [2, 2, 1, 1, 2, 2],
+                ],
+                [1, 1, 2, 2],
```

Comment:
Question: How are you defining "difference" in this PR? If this was set symmetric difference I'd expect this to result in an empty set. Are you defining difference the difference at each index of ordered lists of values? And if that is the case, we aren't really dealing with sets then.

Reply:
And I suppose I have the same question for intersection now that I think about it and looking again at the unit tests in this light.

Reply:
Resolved in another PR (Multisets logic).